### PR TITLE
Add Dawn as an optional build dependency

### PR DESCRIPTION
**Pull request by edevil**
This change adds [Dawn](https://dawn.googlesource.com/) as an optional build dependency. Using the `--//src/workerd/server:enable_webgpu` flag Dawn will be fetched, built, linked and initialised at workerd startup.

There is a separate commit that is the result of running [Buildifier](https://github.com/bazelbuild/buildtools/tree/master/buildifier) on both the WORKSPACE and one of the BUILD files changed.

### CODE REVIEW

File: src/workerd/server/workerd.c++ [hunk 1]
```diff
@@ -20,6 +20,10 @@
 #include <openssl/rand.h>
 #include <workerd/io/compatibility-date.capnp.h>
 
+#ifdef WORKERD_ENABLE_WEBGPU
```
Comment (jasnell) [inline src/workerd/server/workerd.c++:23]: Non-blocking nit... given that this is still experimental/poc, let's make the gate here `WORKERD_EXPERIMENTAL_ENABLE_WEBGPU`... I just want to make it super clear just by looking at it what the status of this is.

File: src/workerd/server/gpu.h [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (jasnell) [inline src/workerd/server/gpu.h:1]: Nit: these files need the copyright boilerplate at the top.

File: src/workerd/server/BUILD.bazel [hunk 3]
```diff
@@ -16,32 +16,48 @@ bool_flag(
...
-    match_all = [":set_use_tcmalloc", ":is_linux"],
+    match_all = [
+        ":set_use_tcmalloc",
+        ":is_linux",
+    ],
+)
+
+# Flag to enable WebGPU support via the Dawn library
+bool_flag(
+    name = "enable_webgpu",
```
Comment (jasnell) [inline src/workerd/server/BUILD.bazel:35]: Nit: `enable_experimental_webgpu`?

File: build/deps/BUILD
Comment (jasnell) [inline build/deps/BUILD:1]: Just curious... why is this file empty?

File: src/workerd/server/gpu.h [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+
+namespace workerd::gpu {
```
Comment (jasnell) [inline src/workerd/server/gpu.h:3]: Non-blocking nit: for someone who may stumble upon this, it's likely worthwhile adding a comment that this is currently experimental and WIP.

**Review by jasnell — APPROVED** [review]
Still have plenty of concerns around the increase to binary size with the new dependency but given that we're in Let's Just See How This Works phase, this totally LGTM with a few minor nits. The nits are mostly about just adding some crystal clear signaling that the status is experimental.

File: build/deps/BUILD
Comment (edevil) [reply build/deps/BUILD:1]: So that Bazel considers this a package and I can refer to the requirements file there. :)

**Pull request merged**